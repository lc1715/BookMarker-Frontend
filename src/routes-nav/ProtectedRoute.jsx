import { useContext } from 'react';
import { Navigate } from 'react-router-dom';
import UserContext from '../auth/UserContext';


function ProtectedRoute({ children }) {
    const { currentUser } = useContext(UserContext);

    if (!currentUser) {
        return <Navigate to='/' />
    };

    return children;
};

export default ProtectedRoute;