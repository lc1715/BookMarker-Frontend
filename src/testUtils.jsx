import UserContext from "./auth/UserContext";

const demoUser = {
    id: 1,
    username: 'testUser',
    email: 'test@email.com',
    volume_ids: ['e3', 't5'],
}

const UserProvider =
    ({ children, currentUser = demoUser, hasSavedBook = () => (true) }) => (
        <UserContext.Provider value={{ currentUser, hasSavedBook }}>
            {children}
        </UserContext.Provider>
    );

export { UserProvider };
