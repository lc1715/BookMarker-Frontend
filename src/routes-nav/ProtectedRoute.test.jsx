import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils";
import ProtectedRoute from "./ProtectedRoute";

it('renders the ProtectedRoute component', () => {
    render(
        <MemoryRouter>
            <UserProvider>
                <ProtectedRoute />
            </UserProvider>
        </MemoryRouter>
    )
})

it('matches with snapshot when logged in', async () => {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <ProtectedRoute />
            </UserProvider>
        </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot();
})

it('matches with snapshot when logged out', function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider currentUser={null}>
                <ProtectedRoute />
            </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});
