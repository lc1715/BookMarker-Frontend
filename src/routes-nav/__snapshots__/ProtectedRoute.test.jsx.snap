// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches with snapshot  when logged in 1`] = `<DocumentFragment />`;

exports[`matches with snapshot 1`] = `<DocumentFragment />`;

exports[`matches with snapshot when logged in 1`] = `<DocumentFragment />`;

exports[`matches with snapshot when logged out 1`] = `<DocumentFragment />`;
